homeassistant/components/tado/__init__.py

```diff
@@ -169,6 +170,33 @@ def __init__(self, hass, username, password, fallback):
             "zone": {},
         }
 
+    def decide_overlay_mode(self, duration, zone_id, overlay_mode=None):
```

Comment:
```suggestion
    def decide_overlay_mode(self, duration: str, zone_id: int, overlay_mode: str | None = None) -> str:
```

Reply:
Also, it would be nice to add tests for this function.

Reply:
Added tests as well